libyaml_q_declaration.md

```diff
@@ -0,0 +1,124 @@
+
+There is no explicit support for any OS platform, however their [Github repository](https://github.com/yaml/libyaml) installation appears to be targeting Linux. The first version of this library was developed in 2006, and it is used widely. There is no explicit metric of how much the library is used, but the equivalent library for python, developed by the same organization is required for at least 150k repositories (According to [Github metrics](https://github.com/yaml/pyyaml/network/dependents?package_id=UGFja2FnZS01MjUyMjEzNQ%3D%3D)) and the `libyaml` library is used for some optional fast functionality. The [safe_yaml](https://rubygems.org/gems/safe_yaml) ruby gem has over 80million downloads and one of its implementations uses `libyaml` through psych. There are various big projects leveraging functionality from the `libyaml` library.
+
+To study the library as an external dependency, it might not be necessary to require high code coverage, but to check if the used parts in the ROS packages are tested, and this is the case for this library.
+
+Considering the previously mentioned reasons, we consider this library to be robust and reliable, and hence we declare it to qualify as a level 1 external dependency.
+
+# Comparison with ROS packages quality standards
+
+## Version policy [1]
+
+### Version Scheme [1.i]
+
+It is not stated if the library supports any kind of version policy
+
+### Version Stability [1.ii]
+    
+Current version is 0.22
+
+### Public API Declaration [1.iii]
+    
+As a C library, elements available in its `yaml.h` are considered to be their public API.
+
+### API Stability Policy [1.iv]
+    
+There is no policy for API stability. This is not a problem because the `libyaml_vendor` package importing the `libyaml dependency` is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+### ABI Stability Policy [1.v]
+    
+There is no policy for ABI stability. This is not a problem because the `libyaml_vendor` package importing the `libyaml` dependency is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+    
+There is not a direct correlation between the `libyaml` releases and the ROS distributions, however this is not a problem because the `libyaml_vendor` package importing the `libyaml` dependency is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+
+## Change Control Process [2]
+
+### Change Requests [2.i]
+    
+Checking through the commits history, it can be seen is not the case.
+
+### Contributor Origin [2.ii]
+    
+Does not have it (or it does not seem like it’s the case)
+
+### Peer Review Policy [2.iii]
+    
+Seems to be followed for pull requests on the github repository, but as not all code changes occur through change requests, this can not be confirmed for these changes.
```

Comment:
```suggestion
Seems to be followed for pull requests on the Github repository, but as not all code changes occur through change requests, this can not be confirmed for these changes.
```